apis/test_config.py

```diff
@@ -100,11 +100,18 @@ class TestConfig(abc.ABC, Generic[A]):
     accelerator: Accelerator type required for this test.
     time_out_in_min: Test timeout in minutes.
     task_owner: Task owner username or link.
+    custom_tpu_name: A custom TPU name.
+    tpu_name_with_suffix: The flag to define if add auto-generated suffix.
+    all_workers: The flag to define if run commands on all workers or worker 0
+      only.
   """
 
   accelerator: A
   time_out_in_min: Optional[int] = attrs.field(default=None, kw_only=True)
   task_owner: str = attrs.field(default='unowned', kw_only=True)
+  custom_tpu_name: str = attrs.field(default='', kw_only=True)
```

Comment:
IMO defining TPU-specific attributes on `TestConfig` isn't the right solution. `TestConfig` should remain general, since it's going to be the parent of the GPU tests as well.

These make the most sense to me as attributes of `TpuTask`, since they're relevant to the case of dynamically creating a QR. They would not be relevant for GKE+TPU VM tests, since GKE controls execution and resource management.

Reply:
Yes, I think this makes more sense as well! Thanks for pointing it out! 

Tested: https://screenshot.googleplex.com/AvguLYU6aAEun6H